lucene/sandbox/src/java/org/apache/lucene/geo/Rectangle2D.java

```diff
@@ -183,6 +191,33 @@ private static Relation compareBBoxToRangeBBox(final byte[] bbox,
         Arrays.compareUnsigned(maxTriangle, maxYOffset, maxYOffset + BYTES, bbox, 2 * BYTES, 3 * BYTES) <= 0) {
       return Relation.CELL_INSIDE_QUERY;
     }
+
```

Comment:
This first check can be skipped for intersects? maybe add it in an else statement